appcli/stack_manager/archive_manager.py

```diff
@@ -0,0 +1,194 @@
+
+
+class ArchiveManager:
+    """
+    Functions to implement archiving rules.
+    """
+
+    def __init__(self, cli_context: CliContext, archives: List = []):
+        """Constructor."""
+        self.cli_context: CliContext = cli_context
+        self.archives: List[ArchiveRuleset] = []
+        # NOTE: Convert to actual objects for validation and easy access.
+        for ruleset in archives:
+            self.archives.append(ArchiveRuleset.model_validate(ruleset))
+
+    def run_all_archive_rulesets(self):
+        """Execute all archive rules."""
+        if len(self.archives) == 0:
+            logger.warn("No rules defined in the stack settings.")
+        for rule in self.archives:
+            self.run_archive_ruleset(rule.name)
+
+    def run_archive_ruleset(self, ruleset_name: str):
+        """Execute a set of archive rules.
+
+        Args:
+          archive_ruleset: Name of the ruleset to execute. Must be present in `self.archives`.
+        """
+        # Bail early if the ruleset does not exist or multiple rulesets with that name are found.
+        matching_rulesets = [
+            ruleset for ruleset in self.archives if ruleset.name == ruleset_name
+        ]
+        if len(matching_rulesets) == 0:
+            raise KeyError(
+                f"The archive rule `{ruleset_name}` was not found in the `stack-settings` file."
+            )
+        if len(matching_rulesets) > 1:
+            raise KeyError(
+                f"Multiple rules matching `{ruleset_name}` were found in the `stack-settings` file."
+            )
+        ruleset: ArchiveRuleset = matching_rulesets[0]
+        logger.info(f"Executing the `{ruleset.name}` archive ruleset.")
+
+        for rule in ruleset.rules:
+            logger.info(f"Executing the `{ruleset.name}.{rule.name}` archiving rule.")
+
+            # Need to determine what type of rule this is.
+            if isinstance(rule, CompressRule):
+                self._run_compress_rule(rule)
+            elif isinstance(rule, PurgeRule):
+                self._run_purge_rule(rule)
+            else:
+                # NOTE: Because pydantic does validation we should not get here.
+                # If this gets triggered, you (the developer) have defined a new `ArchiveRule` without implementing it.
+                # Please fill out the new `elif` branch and corresponding function to resolve.
+                raise NotImplementedError(
+                    f"Rule `{rule.name}` is of type `{type(rule)}` which has not been implemented."
+                )
+
+    def _run_compress_rule(self, rule: CompressRule):
+        """Execute a Compress archive rule."""
+        # Get the name of the archive.
+        dated_archive_file = Path(datetime.datetime.now().strftime(rule.archive_file))
+        archive_path = self.cli_context.data_dir / dated_archive_file
+
+        # Remove the archive file if it already exists.
+        if archive_path.exists():
+            archive_path.unlink()
```

Comment:
We probably need to handle failure, i.e. file could not be deleted for some reason.